Sources/SwiftParser/DelayedParsing.swift

```diff
@@ -0,0 +1,116 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@_spi(RawSyntax) import SwiftSyntax
+
+extension Parser {
+  mutating func skippedBraceBody(unless: Lookahead.SkipBodyResult) -> RawSkippedDeclSyntax? {
+    return self.withLookahead { lookahead in
+      let result = lookahead.advanceUntilMatchingRightBrace()
+      guard result.isDisjoint(with: unless) else {
```

Comment:
Instead of checking the result on the skipped area, the skipping function should bail out and return `nil` as soon as it finds any `unless` condition.